app/src/main/java/com/futurice/freesound/feature/search/SearchViewModel.java

```diff
@@ -96,28 +89,20 @@ void openSoundDetails(@NonNull final Sound sound) {
 
     @Override
     public void bind(@NonNull final CompositeDisposable disposables) {
-        disposables.add(searchTermRelay.subscribeOn(Schedulers.computation())
-                                       .observeOn(Schedulers.computation())
-                                       .map(String::trim)
-                                       .debounce(SEARCH_DEBOUNCE_TIME_SECONDS, TimeUnit.SECONDS)
-                                       .switchMap(getSearchOrClear())
-                                       .observeOn(mainThread())
```

Comment:
Hmm, switching threads to do nothing. Lucky RxJava2 has those performance improvements ;-)